p0937507934/order_system
# [Backend] User api

與這支 PR 有關聯的 Asana 卡片:

URL: [ASANA-Backend](https://app.asana.com/0/1201219031105843/1201219147607889)

# Description

Implement user crud and auth validation.

# Step by Step

Overall we have default superuser and main store below:
```json
{
    "username":"hsipl206",
    "password":"hsipl206",
}
```
```json
{
      "name": "kcy main store",
      "type": 1,
      "status": 0,
      "image": "",
}
```

testflow

login with superuser -> get all store by /api/store to know mainstore id -> create user -> create store -> user login 

1. Create User API -> /api/user/register (Header must use  multipart/form-data)
```
{
    "username":"kcykcy123",
    "password":"hsipl206123",
    "name":"kcygod",
    "type":0,
    "status":0,
    "storeId":1
}
```
2. Update User API -> not implement
3. Login API -> /api/user/login
```
{
   "username": required,
   "password": required
}
```
4. Logout API -> /api/user/logout
5. Auth middleware for check authority


# Screenshot or more (Optional)

-----------------

提交 PR 前，請先檢查以下是否已完成：

* [x] 該分支是從最新 `master` 切出來 (如果不是，請先 rebase/merge 它)
* [ ] 是否新增測試，或被現有的測試覆蓋
* [x] 這支 PR 是否有明確的標題、內容描述

-----------------

補充說明:

1.need to implement update user api
2.still not restrict user get his own data
3.only manager can create his own employee
